python/kafkaopmon/OpMonPublisher.py

```diff
@@ -0,0 +1,120 @@
+        )
+        self.log.info("Initialized Kafka producer")        
+
+    def publish(
+        self,
+        session:str,
+        application:str,
+        message:msg,
+        custom_origin:Optional[dict[str,str]] = {"": ""},
+        substructure:Optional[str] = "",
+    ):
+        """Create an OpMonEntry and send it to Kafka."""
+        t = Timestamp()
+        time = t.GetCurrentTime()
+
+        data_dict = self.map_message(message)
+        opmon_id = OpMonId(
+            session = session,
+            application = application,
+            substructure = substructure
+        )
+        opmon_entry = OpMonEntry(
+            time = time,
+            origin = opmon_id,
+            custom_origin = custom_origin,
+            measurement = message.DESCRIPTOR.name,
+            data = data_dict,
+        )
+
+        target_topic = self.extract_topic(message)
+        target_key = self.extract_key(opmon_entry)
+
+        return self.producer.send(
+            target_topic,
+            value = opmon_entry,
+            key = target_key
+        )
+
+    def extract_topic(self, message:msg) -> str:
+        return self.default_topic
+
+    def extract_key(self, opmon_entry:OpMonEntry) -> str:
+        key = str(opmon_entry.origin.session) 
+
+        if (opmon_entry.origin.application != ""):
+            key += "." + opmon_entry.origin.application
+        for substructureID in opmon_entry.origin.substructure:
+            key += "." + substructureID
+
+        key += '/' + str(opmon_entry.measurement)
+        return key
+
+    def map_message(self, message:msg):
+        message_dict = {}
+        for name, descriptor in message.DESCRIPTOR.fields_by_name.items():
+            if descriptor.cpp_type == fd.CPPTYPE_MESSAGE:
+                message_dict = message_dict | self.map_message(getattr(message, name))
+            else:
+                message_dict[name] = self.map_entry(getattr(message, name), descriptor.cpp_type)
+        return message_dict 
+
+    def map_entry(self, value, field_type:int):
+        formatted_OpMonValue = OpMonValue() 
+        match field_type:
+            case fd.CPPTYPE_INT32:
+                formatted_OpMonValue.int4_value = value
+            case fd.CPPTYPE_INT64:
+                formatted_OpMonValue.int8_value = value
+            case fd.CPPTYPE_UINT32:
+                formatted_OpMonValue.uint4_value = value
+            case fd.CPPTYPE_UINT64:
+                formatted_OpMonValue.uint8_value = value
+            case fd.CPPTYPE_DOUBLE:
+                formatted_OpMonValue.double_value = value
+            case fd.CPPTYPE_FLOAT:
+                formatted_OpMonValue.float_value = value
+            case fd.CPPTYPE_BOOL:
+                formatted_OpMonValue.boolean_value = value
+            case fd.CPPTYPE_STRING:
+                formatted_OpMonValue.string_value = value
+            case _:
+                raise ValueError("Value is of a non-supported type.")
```

Comment:
Technically this is not the behaviour we have in C++. In C++ non supported fields are ignored. 
Then, if the OpMonEntry is completely empty, we throw an error. Actually, in C++ we don't even throw, we just create and ers issue and we print it. They idea being that if something which is not opmon related is worth publishing we can do it directly, without intermediate copies. 

So what to do? What I surely want to is
 - Do NOT publish entry messages
 - Do not throw every time a type is not supported, the real error condition is: empty entry.
 
 If you prefer to throw or just generate an error message when there is in fact an empty entry, I'll leave it to you. I would prefer the idea of the error log other than the throw, the reason being that opmon is a regular operation, you just don't publish a message once, you keep updating values to create time series. And so throwing regularly to me destabilises the system too much.

Reply:
Here
 - we just create and ers issue and we print it